platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/hashmap/Bucket.java

```diff
@@ -419,6 +403,15 @@ private FindResult findEntry(final int keyHashCode, final K key) {
         return FindResult.NOT_FOUND;
     }
 
+    private boolean keyEquals(final long pos, final int size, final Bytes key) {
+        for (int i = 0; i < size; i++) {
+            if (bucketData.getByte(pos + i) != key.getByte(i)) {
+                return false;
+            }
+        }
+        return true;
+    }
```

Comment:
Minor: I'm wondering if this implementation that reads data byte by byte is fast enough, or it may be causing a performance issue of some degree.

Reply:
Yes, I was worried about it, too. Filed a ticket to improve `Bytes.equals()` performance: https://github.com/hashgraph/pbj/issues/278